data/types.go

```diff
@@ -37,15 +39,22 @@ type Signature struct {
 }
 
 type Key struct {
-	Type       string   `json:"keytype"`
-	Scheme     string   `json:"scheme"`
-	Algorithms []string `json:"keyid_hash_algorithms,omitempty"`
-	Value      KeyValue `json:"keyval"`
+	Type       string          `json:"keytype"`
+	Scheme     string          `json:"scheme"`
+	Algorithms []string        `json:"keyid_hash_algorithms,omitempty"`
+	Value      json.RawMessage `json:"keyval"`
 
 	ids    []string
```

Comment:
actually, you have the map from ids to verifiers in the db.keys so again, I am not sure what would be the usage of the ids here.

Reply:
> I am developing a bit of doubt around having ids here. 

The reason why is that so `IDs()` is only calculated once https://github.com/theupdateframework/go-tuf/blob/9545f820e4ee7995841c80d28cbac8c859b233f0/data/types.go#L58-L65 by holding on to the member variable.

> And now without it, Key would be same as PrivateKey.

Actually this is true as a result of the making the `Value` arbitrary dictionaries. But I think for clarity we should keep the Public and Private Keys different -- for a single key type, they respectively serialize the public and private values into those data types. Would it be more clear if I call `Value` `PublicValue` and `PrivateValue` for each of the two types? 

>  I am not sure what would be the usage of the ids here.

Calculating key ID is always done from the key data type since it's a hash of the JSON data of the key -- holding it as a variable is only to stop re-calculation. I don't see how the map from ID to verifiers would remove the need for it though? Since it is the reverse map.

Reply:
if PrivateKey stays, this should be renamed to PublicKey

Reply:
Discussed offline. This will stay in data.PublicKey